package/gluon-mesh-vpn-wireguard/files/lib/gluon/upgrade/404-wireguard-generate-key

```diff
@@ -0,0 +1,6 @@
+#!/bin/sh
+secretfile=/lib/gluon/mesh-vpn/wireguard
+if ! wg pubkey < $secretfile >/dev/null 2>&1
+then
+  wg genkey > $secretfile
```

Comment:
@mortzu (and I) have discovered, that in x86-builds/VMs wireguard blocks booting, because of lake of entropy
on generation private key here.
This is also reseted on a boot / update.

Reply:
Maybe store it boot and update safe in uci under `wireguard.mesh_vpn.secret`.
So that it make just problem during setup.

Reply:
thanks for finding this. i thought this wireguard issue on systems with lack of entropy at boot time was already solved a year ago....

Reply:
@Adorfer do you know more ? does they have installed havaged or so?

Reply:
Other communities add `haveged` to `GLUON_SITE_PACKAGES` in `site.mk`

Reply:
It is also part of the example site.mk https://gluon.readthedocs.io/en/v2018.2.x/user/site.html#examples

Reply:
Okay, is it nessasary only for x86-64 ? or more devices?
I will like to use include it only at some targets

Reply:
many communities add it to every target - but without wireguard, it's only of little use, as far as i remember for ssh and(only?) during config mode.
"add it into docu" - bob already linked the docs?

Reply:
As far as i undestand haveged is good for targets without radio and really needed for those potentially running virtualized.

Reply:
maybe wireguard should have a dependency on haveged? @christf ?

Reply:
@rotanid i meant to docs for this package

Reply:
@genofire this wont be necessary if a dependency is added

Reply:
haveged is useful for the private wifi feature as well.
boot is blocked until enough entropy is available - so the machine does boot. It just takes a lot longer. Imho this should be on the docs because it is possible to use wg without haveged. It just isn't fun.

Reply:
this was added to the docs.